rss/parser.cpp

```diff
@@ -62,51 +62,57 @@ struct HeaderValues {
 	std::string charset;
 
 	HeaderValues()
-		: lastmodified(0)
-		, charset("utf-8")
 	{
+		reset();
+	}
+
+	void reset()
+	{
+		lastmodified = 0;
+		charset = "utf-8";
+		etag = "";
 	}
 };
 
 }
 
 static size_t handle_headers(void* ptr, size_t size, size_t nmemb, void* data)
 {
-	char* header = new char[size * nmemb + 1];
+	const auto header = std::string(reinterpret_cast<const char*>(ptr), size * nmemb);
 	HeaderValues* values = static_cast<HeaderValues*>(data);
 
-	memcpy(header, ptr, size * nmemb);
-	header[size * nmemb] = '\0';
-
-	if (!strncasecmp("Last-Modified:", header, 14)) {
-		time_t r = curl_getdate(header + 14, nullptr);
+	if (header.find("HTTP/") == 0) {
+		// Reset headers if a new response is detected (there might be multiple responses per request in case of a redirect)
+		values->reset();
+	} else if (header.find("Last-Modified:") == 0) {
+		const std::string header_value = header.substr(14);
+		time_t r = curl_getdate(header_value.c_str(), nullptr);
 		if (r == -1) {
 			LOG(Level::DEBUG,
 				"handle_headers: last-modified %s "
 				"(curl_getdate "
 				"FAILED)",
-				header + 14);
+				header_value.c_str());
 		} else {
-			values->lastmodified =
-				curl_getdate(header + 14, nullptr);
+			values->lastmodified = r;
+			curl_getdate(header_value.c_str(), nullptr);
 			LOG(Level::DEBUG,
 				"handle_headers: got last-modified %s (%" PRId64 ")",
-				header + 14,
+				header_value.c_str(),
```

Comment:
`LOG` supports `std::string`, so no need to call `c_str()` here.
```suggestion
				header_value,
```